core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java

```diff
@@ -185,7 +185,7 @@ public void write(String s, byte[] bytes) {
                     TryBlock tryBlock = greaterThan22_2.tryBlock();
 
                     tryBlock.invokeStaticMethod(registerLambdaCapturingClass,
-                            tryBlock.load(i.getClassName()));
+                            tryBlock.loadClassFromTCCL(i.getClassName()));
```

Comment:
Right, this is fixing: https://github.com/graalvm/mandrel/actions/runs/3156296401/jobs/5136091100#step:11:316

OK.